analysis/util/SkipLogic.py

```diff
@@ -0,0 +1,23 @@
+from goratings.interfaces import (
+    GameRecord,
+    Storage,
+)
+
+__all__ = [
+    "should_skip_game",
+]
+
+def should_skip_game(game: GameRecord, storage: Storage) -> bool:
+    ## Only count the first timeout in correspondence games as a ranked loss
+    if game.timeout and game.speed == 3: # correspondence timeout
+        player_that_timed_out = game.black_id if game.black_id != game.winner_id else game.white_id
+        other_player = game.black_id if game.black_id == game.winner_id else game.white_id
+        skip = storage.get_timeout_flag(game.black_id) or self._storage.get_timeout_flag(game.white_id)
```

Comment:
Missed a `self._storage` to `storage` here

Reply:
Fixed in 82a679a3c538c331663028edb43af412babdfb60. Missed this because I was testing with `--aga` for expediency. I've now also tested with `--ogs --games=2000000`.